src/scriptsys/script.c

```diff
@@ -9,12 +9,27 @@
 #include "scr_cmd.h"
 #include "scr_game.h"
 #include "common/profiler.h"
+#include "allocator/allocator.h"
 
 static lua_State* L;
 
+
+static void* scr_lua_alloc(void* ud, void* ptr, size_t osize, size_t nsize)
+{
+	if (nsize == 0)
+	{
+		Mem_Free(ptr);
+		return NULL;
+	}
+	else
+	{
+		return Script_Realloc(ptr, (i32)nsize); // would we ever allocate larger than i32 at once?
```

Comment:
imo if anything is allocating a GB or more its probably doing something wrong, which is why the API uses i32.
Highly recommend more ASSERT() usage, to catch subtle things like this.
```ASSERT((i32)nsize > 0);```

Double check this compiles in Release. You might need to remove ud and osize since they are un-referenced:
```
static void* scr_lua_alloc(void*, void* ptr, usize, usize nsize)
```

Reply:
Yeah I have to get into the habit of using ASSERT more.

Reply:
oops I meant to quote reply but I edited your comment! xD

Reply:
> Double check this compiles in Release. You might need to remove ud and osize since they are un-referenced:
> 
> ```
> static void* scr_lua_alloc(void*, void* ptr, usize, usize nsize)
> ```

Doesn't seem to have problems compiling in release, but I will remove it for cleanliness.

Reply:
Actually I get `unnamed prototyped parameters not allowed when body is present` if I remove the params.